### PR TITLE
Client Conf guide reorganization in alphabetical order, where applicable

### PR DESCRIPTION
**Pull request by 0rnela**
# Description
With the growing lists of supported clients, navigation bar has become very long and difficult to use. It is not easy to scan through it to find the instructions for the specific client. Also, some of the key operations are lost, or not not very prominent, due to the amount of content in the navigation bar (e.g. Delete clients), but the cahnges beyond reordering may be dealt with separately.

I have also discovered an orphan link to file registration-overview.adoc, which is called from many other files, but was not in the nav bar, so I have included it back in.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [x] Manager-4.2  https://github.com/uyuni-project/uyuni-docs/pull/1096
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Fixes #<insert issue or PR link, if any>

### CODE REVIEW

**Review by keichwa — APPROVED** [review]

**Comment by 0rnela** [comment]
Adding PDF to illustrate new nav bar. 

Please pay attention to section Client registration Overview. This page was reinstated to the nav bar and now sits above description of all registration methods, followed by different clients' registrations.

Clients registration files are organised in two parts as two separate levels, overview and actual registration, e.g.:  
*** xref:registration-overview-oracle.adoc[Oracle Client Registration]
**** xref:clients-oracle.adoc[Oracle Clients]

PDFs in general do not show **** level in the nav bar, but since it is nested under *** it is accessible that way.

HTML files can show both **** and *** in the nav bar, but I find PDF acceptable as it is.


Unless there are further issues with PDF nesting, this PR is now ready for full review and I am removing [WIP] from the title.

[PR1068_uyuni_client-configuration_guide.pdf](https://github.com/uyuni-project/uyuni-docs/files/6915529/PR1068_uyuni_client-configuration_guide.pdf)

**Review by jcayouette — APPROVED** [review]

**Comment by juliogonzalez** [comment]
@0rnela for the navigation bar review, I'd really prefer having the HTML data. Just build it, generate a tarball, and attach it. It's way easier to review than the PDF :\

**Comment by 0rnela** [comment]
@juliogonzalez -- thanks for the tip, here are just HTMLs.

[uyuni-guides-noPDFs.tar.gz](https://github.com/uyuni-project/uyuni-docs/files/6917784/uyuni-guides-noPDFs.tar.gz)

**Comment by juliogonzalez** [comment]
Thanks! Reviewing now :-)

File: modules/client-configuration/nav-client-configuration-guide.adoc [hunk 1]
```diff
@@ -40,59 +41,61 @@ endif::[]
...
+**** xref:registration-bootstrap.adoc[Register with Bootstrap]
+**** xref:registration-cli.adoc[Register on the Command Line]
 // SUSE Client Registration
-** xref:registration-overview-suse.adoc[SUSE Client Registration]
-*** xref:clients-sle.adoc[SUSE Linux Enterprise Clients]
-*** xref:clients-opensuse.adoc[openSUSE Clients]
-*** xref:clients-sleses.adoc[Expanded Support Clients]
+*** xref:registration-overview-suse.adoc[SUSE Client Registration]
+**** xref:clients-sle.adoc[SUSE Linux Enterprise Clients]
+**** xref:clients-opensuse.adoc[openSUSE Clients]
```
Comment (juliogonzalez) [inline modules/client-configuration/nav-client-configuration-guide.adoc:54]: Can we take the opportunity to remove openSUSE and openSUSE MicroOS as SUSE clients, and instead create `openSUSE Clients`, and then have `openSUSE Leap` and `openSUSE MicroOS`?

I only see benefits with this: openSUSE is easier to find and we clearly show what we support (Leap and MicroOS, not tumbleweed, at least not for now).

File: modules/client-configuration/nav-client-configuration-guide.adoc [hunk 2]
```diff
@@ -40,59 +41,61 @@ endif::[]
...
+*** xref:registration-methods.adoc[Registration Methods]
+**** xref:registration-webui.adoc[Register with the Web UI]
+**** xref:registration-bootstrap.adoc[Register with Bootstrap]
+**** xref:registration-cli.adoc[Register on the Command Line]
 // SUSE Client Registration
-** xref:registration-overview-suse.adoc[SUSE Client Registration]
-*** xref:clients-sle.adoc[SUSE Linux Enterprise Clients]
-*** xref:clients-opensuse.adoc[openSUSE Clients]
-*** xref:clients-sleses.adoc[Expanded Support Clients]
+*** xref:registration-overview-suse.adoc[SUSE Client Registration]
```
Comment (juliogonzalez) [inline modules/client-configuration/nav-client-configuration-guide.adoc:52]: What do you think we remove "Client Registration" and we leave only the OS/Vendor name in all cases? In the end we are already inside the "Client registration" section.

**Review by juliogonzalez — COMMENTED** [review]
Looks good in general, but I left some suggestions.

**Comment by keichwa** [comment]
1/ I think we need a changelog entry.

2/ I'm wondering whether we actually want to have it backported to 4.2?

**Comment by 0rnela** [comment]
> 1/ I think we need a changelog entry.
> 
> 2/ I'm wondering whether we actually want to have it backported to 4.2?

Thankd @keichwa. I'll add changelog.

I see value in backporting to 4.2 as we have just released it and easier to navigate guide is always a benefit in my view, but we can raise this at the standup and make a decision with bigegr pannel.

File: modules/client-configuration/nav-client-configuration-guide.adoc (hunk 2 above)
Comment (0rnela) [reply modules/client-configuration/nav-client-configuration-guide.adoc:52]: > What do you think we remove "Client Registration" and we leave only the OS/Vendor name in all cases? In the end we are already inside the "Client registration" section.

I'm glad you asked @juliogonzalez, I've been thinking along similar lines as I'm always in favour of simplifying and decluttering. I'll speak with the team and would in any case deal with this in separate PR.

**Comment by 0rnela** [comment]
@juliogonzalez -- this addresses the changes about splitting SUSE from openSUSE clients if I understood you correctly. If yes, I'd also move openSUSE furyher down the navigation bar, to match alphabetical order.

**Comment by 0rnela** [comment]
[suseClientsReorganized.tar.gz](https://github.com/uyuni-project/uyuni-docs/files/6922986/suseClientsReorganized.tar.gz)

@juliogonzalez -- this addresses the changes about splitting SUSE from openSUSE clients if I understood you correctly.

The other change about less pages per client would be dealt with separately.

**Comment by 0rnela** [comment]
[suseClientsReorganized3.tar.gz](https://github.com/uyuni-project/uyuni-docs/files/6923380/suseClientsReorganized3.tar.gz)


@juliogonzalez Starting with SUSE and openSUSE, followed by all other clients alphabetized.

Final comments please @keichwa and @jcayouette?

**Review by keichwa — APPROVED** [review]

**Review by juliogonzalez — APPROVED** [review]
LGTM

**Pull request merged**